src/txmempool.h

```diff
@@ -319,6 +321,18 @@ class CInPoint
     size_t DynamicMemoryUsage() const { return 0; }
 };
 
+/** Reason why a transaction was removed from the mempool,
+ * this is passed to the notification signal.
+ */
+enum MemPoolRemovalReason {
```

Comment:
This enum seems to be part of the ZMQ API, so we'll probably want some stability across versions. Things like MPR_EXPIRY and MPR_SIZELIMIT seem quite specific to the existing implementation.